### DISCUSSION TITLE
Initialise GP without any training data.

### DISCUSSION THREAD

**Discussion by yaswanth1701** [cornellius-gp/gpytorch#2485] [Q&A] [answered]
Hi everyone, 
just a small question, is there a way to initialise the exact GP without any training data ? I want to make prediction on test data, without conditioning it on any training data (similar to Sklearn GP). Few more things I would like to mention: 1) there is no training involved in my case only conditioning the GP based on training data with set hyper-parameters 2) and I am using GP for path planning and training data added iteratively to GP but for the first iteration I don't have any data to condition on however I require prediction on test data for the first iteration.

**Comment by jacobrgardner** [comment, marked as answer]
Both `train_inputs` and `train_targets` can be `None`

**Comment by yaswanth1701** [comment]
Thanks a lot for your answer, I previously tried running it with values set to `None`  but it wasn't working well however I retried running it with a few changes and it works now !!. One more thing I wanted to ask I am currently focussed on increasing the speed of prediction (both mean and variance) for test data as much as possible as compared to Sklearn GP. I have to add new data to the model iteratively, for which there seem to exist two ways `get_fantasy_model` (adds training data to the model in linear time given data size is small which is my use case) and `set_train_data` (replaces the whole training data with a new one) the catch here is I want to remove the additional data from the model also after every iteration. So, is there a way to do this efficiently instead of using `set_train_data` method every time (something like `get_fantasy_model` but for removing training data from the model)?
